chromadb/utils/embedding_functions.py

```diff
@@ -334,10 +342,27 @@ def _init_model_and_tokenizer(self) -> None:
             # https://github.com/UKPLab/sentence-transformers/blob/3e1929fddef16df94f8bc6e3b10598a98f46e62d/docs/_static/html/models_en_sentence_embeddings.html#LL480
             self.tokenizer.enable_truncation(max_length=256)
             self.tokenizer.enable_padding(pad_id=0, pad_token="[PAD]", length=256)
+
+            if self._preferred_providers is None or len(self._preferred_providers) == 0:
+                if len(self.ort.get_available_providers()) > 0:
+                    logger.warning(
```

Comment:
I don't think we should warn here, as now it will always warn by default and I think this will be confusing to users.

Reply:
Ok, but can we at least have a debug message in case someone needs to dig deeper with debug logging?